src/components/promoCode/services/promoCodeSrv.service.js

```diff
@@ -2,8 +2,15 @@
     'use strict';
 
     angular.module('znk.infra-web-app.promoCode').provider('PromoCodeSrv',
-        function () {
+        function (ENV) {
+
             var backendData = {};
+            var appContext = ENV.firebaseAppScopeName;
+
+            backendData[appContext] = {  //default data
```

Comment:
I don't understand why you need this obj map ? in infra-web-app you automaticlly get ENV params based on your existing app context, no need to get the app....
or I'm missing something